rust/src/lib/ifaces/bond.rs

```diff
@@ -412,6 +418,33 @@ impl BondInterface {
         }
         ret
     }
+
+    pub(crate) fn resolve_ports_ref_by_profile_name(
+        &mut self,
+        profile2iface: &Profile2IfaceName,
+    ) -> Result<(), NmstateError> {
+        if let Some(ports_config) = self
+            .bond
+            .as_mut()
+            .and_then(|b| b.ports_config.as_deref_mut())
+        {
+            for port_conf in
+                ports_config.iter_mut().filter(|p| p.name.is_empty())
```

Comment:
Can we raise an error when both `name` and `profile_name` are defined?

Reply:
we cannot. Nmstate allows applying back the queried state, in queried state we include both profile_name and interface name.

Reply:
Schema changed. Please check the patch again.